dlt/destinations/sql_jobs.py

```diff
@@ -501,6 +501,12 @@ def gen_merge_sql(
             root_table["name"]
         )
 
+        # NOTE: this is bigquery specific code! Move to bigquery merge job
+        # NOTE: we also need to create all child tables
+        # NOTE: this will not work if the schema of the staging table changes in the next run..
+        # in some cases we need to create final tables here
+        sql.append(f"CREATE TABLE IF NOT EXISTS {root_table_name} LIKE {staging_root_table_name};")
```

Comment:
this code should be executed at the end of each copy job - and only if autodetect schema is enabled. this way we can just run our merge job as usual.

we could also migrate columns from staging table to destination tables but maybe we can start with just cloning

Reply:
I have prepended it so the bigquery merge jobs for autodetect tables now. The other solution would be much more complicated with two consecutive followupjobs. I can do it that way too, but I think my solution is fairly clean...